### DISCUSSION TITLE
Grouchat of 7 Agents and Groupchat manager as Coordinator

### DISCUSSION THREAD

**Discussion by ArtDu030** [microsoft/autogen#2943] [General]
Hi there,

I've been experimenting with creating a group chat involving 7 agents. Each agent is supposed to give their feedback, which should then be summarized by the Coordinator. However, I haven't been able to verify if this summarization works correctly. Here's what I've done so far: I initialized the agents and the group chat using a local model with LM Studio. Occasionally, when I start the conversation, it stops immediately with an "content field is empty" error. But when I reset the model and restart the server, it works. Here’s my current progress:

    I (userproxy) initiate a conversation with input in the chat.
    This input is sent to the Coordinator, who forwards it to the first agent (Analyzer).
    The Analyzer processes the content and is supposed to forward it to the Coordinator, who should then forward it to the Communicator agent. However, it stops at this point.

I configured the Analyzer agent in the system message to start its content with "Forward to Communicator". My Coordinator is configured as follows:

Steps to Follow:

    Delegate the analysis task to the Analyzer.
    If you receive analysis content from the Analyzer, forward it to the Communicator.
    If you receive analysis from the Communicator, then send the analysis from the Analyzer to the Activity_Coordinator.
    If you receive analysis from the Activity_Coordinator, then send the analysis from the Analyzer to the Boundary_Setter.
    If you receive analysis from the Boundary_Setter, then send the analysis from the Analyzer to the Conflict_Resolver.
    If you receive analysis from the Conflict_Resolver, then send the analysis from the Analyzer to Patience_and_Persistence.
    If you receive analysis from Patience_and_Persistence, then summarize the analysis from all agents and send it back to the userproxy.
    Reply 'TERMINATE' at the end when all tasks are done.

In the server logs, I see a role labeled "Assistant" with empty content. I don't know where this role "Assistant" comes from since I haven't defined it in the group chat. Does anyone have any ideas?

In summary: Can anyone with experience tell me how I can forward the content from one agent to another through the Coordinator? Also, does anyone know why the role "Assistant" appears in the logs when I haven't defined such a role?

I found .py files in \Lib\site-packages\flaml\autogen\agentchat but dont know really if a modification is needed to solve my issue if there is any. 

Here's the log content before the error occurs:
 {
      "content": "",
      "role": "assistant"
    },
    {
      "content": "You didn't choose a speaker. As a reminder, to determine the speaker use these prioritised rules:\n    1. If the context refers to themselves as a speaker e.g. \"As the...\" , choose that speaker's name\n    2. If it refers to the \"next\" speaker name, choose that name\n    3. Otherwise, choose the first provided speaker's name in the context\n    The names are case-sensitive and should not be abbreviated or changed.\n    The only names that are accepted are ['Analyzer', 'Communicator', 'Activity_Coordinator', 'Boundary_Setter', 'Conflict_Resolver', 'Patience_and_Persistence', 'External_Help'].\n    Respond with ONLY the name of the speaker and DO NOT provide a reason.",
      "role": "system"
    }
  ],
  "model": "llama2",
  "max_tokens": 3000,
  "stream": false,
  "temperature": 0.1
}
[2024-06-14 16:59:31.162] [ERROR] [Server Error] {"title":"'messages' array must only contain objects with a 'content' field that is not empty"}

**Comment by ghost** [comment]


**Reply by ArtDu030** [reply]
If you cant contribute then please dont spam this here.

**Comment by marklysze** [comment]
Hi @ArtDu030, I've noticed blank responses with LM Studio when it receives messages whereby the last message has a role "system" (or possibly "assistant", too. I believe it returns blank because it thinks that it's finishing off the system message, which is already complete.

Can you check if that's the case and let me know. I believe we can surpress the system message that shows at the end.

**Reply by CrosswiredStudios** [reply]
This is what is happening to me.

[2024-07-12 12:00:19.855] [INFO] Received POST request to /v1/chat/completions with body: {
  "messages": [
    {
      "content": "You are in a role play game. The following roles are available:\n                gitlab_assistant: Charter Gitlab Assistant\nanalysis_assistant: GitLab Error Analysis Assistant.\n                Read the following conversation.\n                Then select the next role from ['gitlab_assistant', 'analysis_assistant'] to play. Only return the role.",
      "role": "system"
    },
    {
      "content": "please get the next gitlab error",
      "role": "user",
      "name": "user_proxy"
    },
    {
      "content": "Please provide the JSON object containing the `id` property and the `errorLog` property, and I'll analyze it for you.\n\n(Note: You can paste the JSON object here, and I'll respond with the analysis)",
      "role": "user",
      "name": "analysis_assistant"
    },
    {
      "content": "Read the above conversation. Then select the next role from ['gitlab_assistant', 'analysis_assistant'] to play. Only return the role.",
      "role": "system"
    },
    {
      "content": "",
      "role": "assistant"
    },
    {
      "content": "You didn't choose a speaker. As a reminder, to determine the speaker use these prioritised rules:\n    1. If the context refers to themselves as a speaker e.g. \"As the...\" , choose that speaker's name\n    2. If it refers to the \"next\" speaker name, choose that name\n    3. Otherwise, choose the first provided speaker's name in the context\n    The names are case-sensitive and should not be abbreviated or changed.\n    The only names that are accepted are ['gitlab_assistant', 'analysis_assistant'].\n    Respond with ONLY the name of the speaker and DO NOT provide a reason.",
      "role": "system"
    }
  ],
  "model": "llama3:latest",
  "max_tokens": 4000,
  "stream": false,
  "temperature": 0.1
}
[2024-07-12 12:00:19.856] [ERROR] [Server Error] {"title":"'messages' array must only contain objects with a 'content' field that is not empty"}

**Reply by marklysze** [reply]
On your groupchat object can you set role_for_select_speaker_messages to 'user'. It looks like the model is trying to complete the last system message and as it is essentially complete it returns nothing.

Let us know how you go.

**Reply by CrosswiredStudios** [reply]
"user" did seem to work. I ran into some new issues, but this moves it along - thanks again for that insight.

Also its worth mentioning that in AutogenStudio, this is happening behind the scenes and theres no way for a person to set the value to user.

I now have the code
`
import autogen

config_list_lmstudio_llama3 = [
    {
        "model": "lmstudio-community/Meta-Llama-3-8B-Instruct-GGUF",
        "base_url": "http://localhost:1234/v1",
        "api_key": "lm-studio" 
    }
]
llm_config={
    "config_list": config_list_lmstudio_llama3,
}
llm_config = {"config_list": config_list_lmstudio_llama3, "cache_seed": 42}
user_proxy = autogen.UserProxyAgent(
    name="User_proxy",
    system_message="A human admin.",
    code_execution_config={
        "last_n_messages": 2,
        "work_dir": "groupchat",
        "use_docker": False,
    },  # Please set use_docker=True if docker is available to run the generated code. Using docker is safer than running the generated code directly.
    human_input_mode="TERMINATE",
)
coder = autogen.AssistantAgent(
    name="Coder",
    llm_config=llm_config,
)
pm = autogen.AssistantAgent(
    name="Product_manager",
    system_message="Creative in software product ideas.",
    llm_config=llm_config,
)

groupchat = autogen.GroupChat(
    agents=[user_proxy, coder, pm], 
    role_for_select_speaker_messages="user",
    messages=[], max_round=12)
manager = autogen.GroupChatManager(groupchat=groupchat, llm_config=llm_config)

user_proxy.initiate_chat(
    manager, message="Find a latest paper about gpt-4 on arxiv and find its potential applications in software."
)
`

which gives the output

[2024-07-12 14:14:22.773] [INFO] [LM STUDIO SERVER] Processing queued request...
[2024-07-12 14:14:22.773] [INFO] Received POST request to /v1/chat/completions with body: {
  "messages": [
    {
      "content": "You are in a role play game. The following roles are available:\n                User_proxy: A user that can run Python code or input command line commands at a Linux terminal and report back the execution results.\nCoder: A helpful and general-purpose AI assistant that has strong language skills, Python skills, and Linux command line skills.\nProduct_manager: Creative in software product ideas..\n                Read the following conversation.\n                Then select the next role from ['User_proxy', 'Coder', 'Product_manager'] to play. Only return the role.",
      "role": "system"
    },
    {
      "content": "Find a latest paper about gpt-4 on arxiv and find its potential applications in software.",
      "role": "user",
      "name": "User_proxy"
    },
    {
      "content": "A great task! ... I hope this summary and potential applications in software have been helpful!",
      "role": "user",
      "name": "Coder"
    },
    {
      "content": "",
      "role": "user",
      "name": "Coder"
    },
    {
      "content": "Read the above conversation. Then select the next role from ['User_proxy', 'Coder', 'Product_manager'] to play. Only return the role.",
      "role": "user"
    }
  ],
  "model": "lmstudio-community/Meta-Llama-3-8B-Instruct-GGUF",
  "stream": false
}
[2024-07-12 14:14:22.773] [ERROR] [Server Error] {"title":"'messages' array must only contain objects with a 'content' field that is not empty"}
